pkg/api/meta/interfaces.go

```diff
@@ -144,6 +144,7 @@ type RESTMapping struct {
 // consumers of Kubernetes compatible REST APIs as defined in docs/api-conventions.md.
 type RESTMapper interface {
 	VersionAndKindForResource(resource string) (defaultVersion, kind string, err error)
```

Comment:
There should be a comment describing the relationship between groups, resources, versions, and kinds and this seems like a logical place to do it.  These interface methods are all assuming that `resource` identifies version, kind, and group, but that means that resources must be unique across groups.  That would seem to partially defeat the purpose of the groups.